### ISSUE TITLE
Adds graalvm

### ISSUE THREAD

**Issue by bric3** [halcyon/asdf-java#61]
Another idea, ship graalvm distributions.
Ideally they should be separated by their language support. E.g starting from graalvm 19.3, graalvm comes with Java 8 and Java 11.

**Comment by x80486** [comment]
GraalVM has its own plugin, right? Is this different from that or the same thing?

**Comment by bric3** [comment]
Yes it does, actually I wanted to regroup java distribution management. But I understand the point.

**Comment by bric3** [comment]
Another point to account is that if one is using java and graalvm plugin together, they can have two version of jdk binaries at the same time, like `java` which may trigger some issues. Only having a single place to choose which jdk seems safer.

**Comment by x80486** [comment]
What I was trying to say is that, if this work is the same (or very similar) to what the GraalVM plugin does, it should be done in the repo for that plugin. Incompatibilities between versions is something developers would need to tackle by using and changing back and forth the `.tool-versions` file.

I don't think this repo should install anything other than Java. Putting GraalVM in the equation is no different than adding also Kotlin, Scala, and all those tools that requires Java.

On the other hand, I'm pretty bad in shell scripting (I rarely do any shell script); I barely understand what you wrote there :rofl:

**Comment by bric3** [comment]
I just tried having java and graalvm plugin, I just can't use graalvm, because java plugin always wins (maybe because `j` is after `g`).

This is dance is more complicated in my opinion with both graalvm plugin and java plugin, that if a single java plugin is used. By the way I'm not a fan of sdkman for some aspects, but I think they got this aspect right, e.g. they list graalvm as yet another java distribution.

```
❯ asdf global graalvm 20.0.0-java11
❯ asdf current
graalvm        20.0.0-java11 (set by /Users/bric3/.tool-versions)
java           amazon-corretto-11.0.6.10.1-2 (set by /Users/bric3/.tool-versions)
❯ asdf which java
/Users/bric3/.asdf/installs/java/amazon-corretto-11.0.6.10.1-2/bin/java
```

**Comment by cricket007** [comment]
I wonder where SDKMan gets it's sources

```
$ sdk list java | grep grl
 GraalVM       |     | 20.0.0.r11   | grl     |            | 20.0.0.r11-grl
               |     | 20.0.0.r8    | grl     |            | 20.0.0.r8-grl
               |     | 19.3.1.r11   | grl     |            | 19.3.1.r11-grl
               |     | 19.3.1.r8    | grl     |            | 19.3.1.r8-grl
               |     | 19.3.0.r11   | grl     |            | 19.3.0.r11-grl
               |     | 19.3.0.r8    | grl     |            | 19.3.0.r8-grl
               |     | 19.3.0.2.r11 | grl     |            | 19.3.0.2.r11-grl
               |     | 19.3.0.2.r8  | grl     |            | 19.3.0.2.r8-grl
               |     | 19.2.1       | grl     |            | 19.2.1-grl
               |     | 19.1.1       | grl     |            | 19.1.1-grl
               |     | 19.0.2       | grl     |            | 19.0.2-grl
               |     | 1.0.0        | grl     |            | 1.0.0-rc-16-grl
```

**Comment by bric3** [comment]
It puzzled me too, after a bit of investigation and bug reporting I found that Sdkman has everything in a mongodb database, that is now generated / updated using what they call _db migration scripts_ (written in scala).

References to graal appears in the [generic java migration script](https://github.com/sdkman/sdkman-db-migrations/blob/master/src/main/scala/io/sdkman/changelogs/JavaMigrations.scala#L246-L346) and in a [specific graalvm migration script](https://github.com/sdkman/sdkman-db-migrations/blob/master/src/main/scala/io/sdkman/changelogs/java/GraalVmMigrations.scala).

So sdkman actually explicitly code the actual the sdk download source, and it's updated manually (via PR eventually), rather than using generic code like this project that uses the github api.

**Comment by cricket007** [comment]
Interesting. Yeah, if the source can be queried directly, I'd like to think that's cheaper and better overall

**Comment by thuandt** [comment]
@bric3 I was think about using sdkman API to download java and GraalVM. I think here is API reference: https://github.com/sdkman/sdkman-candidates/blob/master/conf/routes . It's simple Play Scala app.

Then: 
+ List all latest java for Linux: `https://api.sdkman.io/2/candidates/java/Linux64/versions/list?current=&installed=`
+ Download link for GraalVM 20.0.0 java8 `https://api.sdkman.io/2/broker/download/java/20.0.0.r8-grl/Linux64`

**Comment by bric3** [comment]
@thuandt Sorry for the late reply, actually I'm not a fan to rely on sdkman api, as we become dependent on their API both in availability and contract.

**Comment by thuandt** [comment]
This issue should be closed in favor of #87

**Comment by bric3** [comment]
Now that #87 is merged, this issue can be closed indeed

```
$ asdf list-all java
...
graalvm-19.0.0
graalvm-19.0.2
graalvm-19.1.0
graalvm-19.1.1
graalvm-19.2.0
graalvm-19.2.0.1
graalvm-19.2.1
graalvm-19.3.0+java8
graalvm-19.3.0+java11
graalvm-19.3.0.2+java8
graalvm-19.3.0.2+java11
graalvm-19.3.1+java8
graalvm-19.3.1+java11
graalvm-19.3.2+java8
graalvm-19.3.2+java11
graalvm-20.0.0+java8
graalvm-20.0.0+java11
graalvm-20.1.0+java8
graalvm-20.1.0+java11
...
```
